crates/uv-auth/src/keyring.rs

```diff
@@ -0,0 +1,103 @@
+use std::process::Command;
+
+use thiserror::Error;
+use tracing::debug;
+use url::Url;
+
+use crate::store::{BasicAuthData, Credential};
+
+/// Keyring provider to use for authentication
+///
+/// See <https://pip.pypa.io/en/stable/topics/authentication/#keyring-support>
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "clap", derive(clap::ValueEnum))]
+pub enum KeyringProvider {
+    // Will not use keyring for authentication
+    #[default]
+    Disabled,
+    // Will use keyring CLI command for authentication
+    Subprocess,
+    // Auto, - not yet implemented
+    // Import, - will probably never be implemented
+}
```

Comment:
> Do you think that it's worth using something like [docs.rs/keyring/latest/keyring](https://docs.rs/keyring/latest/keyring/) instead of a subprocess?

Instead?  No.  In addition?  Yes.

As mentioned in issue https://github.com/astral-sh/uv/issues/1520 - the cloud providers provide backend plugins for the python keyring tool.
Using something like https://docs.rs/keyring/latest/keyring/ would be fine for the `Import` case, but supporting a command-line `keyring` should still exist regardless (especially if you're trying to keep parity with pip) and if someone makes a CLI keyring with cloud provider support based on `keyring-rs` in the future, cool.  If it has the same CLI interface, it should just work.  If it has a different name, add to the `KeyringProvider`s or implement the custom version like discussed with @vlad-ivanov-name above.

Reply:
👍 makes sense